lib/rules/no-multi-comp.js

```diff
@@ -61,21 +103,46 @@ module.exports = {
       );
     }
 
-    return {
+    /**
+     * Checks if the component is exported, if exportOnly is set
+     * @param {Object} component The component being checked.
+     * @returns {boolean} True if the component is exported or exportOnly is false
+     */
+    function isPrivate(component) {
+      return ignorePrivate && exportedComponents.has(findComponentIdentifierFromComponent(component));
+    }
+
+    const rule = {
       'Program:exit'() {
         if (components.length() <= 1) {
           return;
         }
 
         values(components.list())
           .filter((component) => !isIgnored(component))
+          .filter((component) => !isPrivate(component))
           .slice(1)
           .forEach((component) => {
-            report(context, messages.onlyOneComponent, 'onlyOneComponent', {
-              node: component.node,
-            });
+            report(context,
+              ignorePrivate ? messages.onlyOneExportedComponent : messages.onlyOneComponent,
+              ignorePrivate ? 'onlyOneExportedComponent' : 'onlyOneComponent',
+              {
+                node: component.node,
+              });
           });
       },
     };
+
+    if (ignorePrivate) {
+      rule.ExportNamedDeclaration = (node) => {
+        exportedComponents.add(getExportedComponentName(node));
+      };
+
+      rule.ExportDefaultDeclaration = (node) => {
+        exportedComponents.add(getExportedComponentName(node));
+      };
+    }
```

Comment:
instead of mutating maybe we could use Object.assign here?

Reply:
Yeah sure, sounds good. I'm not clear on why but I'm not super experienced in the eslint lifecycle and I can imagine that mutation may introduce issues somehow.

Reply:
it probably doesn't matter much at all, but mutation generally risks slowdowns later, as opposed to creating objects all at once.

Reply:
Switched over.